src/patterns/ec2-app/base.ts

```diff
@@ -62,6 +63,45 @@ export interface Alarms {
   noMonitoring?: false;
 }
 
+/*
+ * GoogleAuthProps adds Google Auth at the ALB level using OIDC.
+ *
+ * For this to work you will need a suitable Google Project with credentials,
+ * which must be added to the relevant locations (see props for details). In
+ * addition, you MUST set 'User Type' to 'Internal' in the Google Console Consent
+ * Screen to ensure only @guardian.co.uk members can access your app.
+ *
+ * To access the user's email information, for example to check Google group
+ * membership or otherwise narrow access, validate the JWT token in the
+ * x-amzn-oidc-data` header and access the claims. To do this, you MUST check
+ * that the token is a valid JWT token - specifically, that it was encoded using
+ * the `ES256` algorithm and that the contained `email` claim is a
+ * @guardian.co.uk email address. See
+ * https://docs.aws.amazon.com/elasticloadbalancing/latest/application/listener-authenticate-users.html#user-claims-encoding
+ * for some example code.
+ *
+ * WARNING: any access to your EC2 instances that does not go via the ALB will
+ * *bypass* the Google auth. Typically you should ensure that your EC2 security
+ * groups only allow access via the ALB (this pattern's default behaviour).
+ *
+ * For more information, see:
+ * https://docs.aws.amazon.com/elasticloadbalancing/latest/application/listener-authenticate-users.html.
+ *
+ */
+export interface GoogleAuthProps {
+  enabled: true;
+
+  // Parameter Store path containing your Google Client ID. Defaults to
+  // `/:STAGE/:stack/:app/googleClientId`.
+  clientIdPath?: string;
+
+  // *Secrets Manager* (NOT Parameter Store) path containing your Google Client
+  // Secret. Defaults to `/:STAGE/:stack/:app/googleClientSecret`. Nb. you
+  // should store your secret as plaintext rather than key-value structured data
+  // when creating it.
+  clientSecretPath?: string;
```

Comment:
With our current restrictions in Google Cloud, creation of a new project is manual and non-trivial. Additionally, creating this Secret Manager value is manual.

I think we can significantly improve the DX here with a combination of [terraform](https://github.com/hashicorp/terraform-provider-google/issues/6074) and [custom types](https://github.com/iann0036/cfn-tf-custom-types) to create the Google project with CloudFormation. It should also be possible to make the Google Project `internal` too.